Sentinel does not respect NatMapper updated hostname

### Redis version

Valkey 8.0.2

### Redisson version

3.43.0

### Redisson configuration

I have a 3-node Valkey cluster running with Sentinel in Kubernetes. I am trying to connect to the cluster from my local machine by port-forwarding all 3 nodes locally.

| Kubernetes hostname | Local Redis port | Local Sentinel port |
|--------|--------|--------|
| valkey-node-0.valkey-headless.default.svc.cluster.local | localhost:1110 | localhost:2110 |
| valkey-node-1.valkey-headless.default.svc.cluster.local | localhost:1111 | localhost:2111 |
| valkey-node-2.valkey-headless.default.svc.cluster.local | localhost:1112 | localhost:2112 |

```java
Config config = new Config();
config.useSentinelServers().setMasterName("myprimary")
    .setPassword("****")
    .setNatMapper(new NatMapper() {

        @Override
        public RedisURI map(RedisURI uri) {
            if (uri.getHost().endsWith("svc.cluster.local")) {
                System.out.println("Mapping: " + uri);
                int port = 1110;
                if (uri.getPort() == 26379) port += 1000;
                if (uri.getHost().startsWith("valkey-node-0")) port += 0;
                if (uri.getHost().startsWith("valkey-node-1")) port += 1;
                if (uri.getHost().startsWith("valkey-node-2")) port += 2;
                uri = new RedisURI(uri.getScheme(), "localhost", port);
                System.out.println("Re-mapped: " + uri);
            }
            return uri;
        }
        
    })
    .addSentinelAddress("redis://127.0.0.1:2110", "redis://127.0.0.1:2111", "redis://127.0.0.1:2112");
redisson = Redisson.create(config);
```

### What is the Expected behavior?

Redisson should respect the `NatMapper` configuration in all connections to the Sentinel ports.

### What is the Actual behavior?

In some cases, the NatMapper is ignored and Redisson attempts to resolve `redis://valkey-node-2.valkey-headless.default.svc.cluster.local:6379`, which fails.

Full log:

```
2025-01-19T11:44:19.640-05:00  INFO 9959 --- [demo] [           main] org.redisson.Version                     : Redisson 3.43.0
Mapping: redis://valkey-node-2.valkey-headless.default.svc.cluster.local:6379
Re-mapped: redis://localhost:1112
2025-01-19T11:44:19.763-05:00  INFO 9959 --- [demo] [           main] o.r.c.SentinelConnectionManager          : master: localhost/127.0.0.1:1112 added
Mapping: redis://valkey-node-0.valkey-headless.default.svc.cluster.local:6379
Re-mapped: redis://localhost:1110
2025-01-19T11:44:19.766-05:00  INFO 9959 --- [demo] [           main] o.r.c.SentinelConnectionManager          : slave: localhost/127.0.0.1:1110 added
Mapping: redis://valkey-node-1.valkey-headless.default.svc.cluster.local:6379
Re-mapped: redis://localhost:1111
2025-01-19T11:44:19.767-05:00  INFO 9959 --- [demo] [           main] o.r.c.SentinelConnectionManager          : slave: localhost/127.0.0.1:1111 added
Mapping: redis://valkey-node-2.valkey-headless.default.svc.cluster.local:26379
Re-mapped: redis://localhost:2112
Mapping: redis://valkey-node-1.valkey-headless.default.svc.cluster.local:26379
Re-mapped: redis://localhost:2111
2025-01-19T11:44:19.788-05:00  INFO 9959 --- [demo] [sson-netty-1-14] o.r.c.SentinelConnectionManager          : sentinel: redis://127.0.0.1:2112 added
2025-01-19T11:44:19.788-05:00  INFO 9959 --- [demo] [sson-netty-1-15] o.r.c.SentinelConnectionManager          : sentinel: redis://127.0.0.1:2110 added
2025-01-19T11:44:19.794-05:00  INFO 9959 --- [demo] [sson-netty-1-16] o.r.c.SentinelConnectionManager          : sentinel: redis://127.0.0.1:2111 added
2025-01-19T11:44:19.813-05:00  INFO 9959 --- [demo] [sson-netty-1-22] o.redisson.connection.ConnectionsHolder  : 1 connections initialized for 127.0.0.1/127.0.0.1:1112
2025-01-19T11:44:19.954-05:00  INFO 9959 --- [demo] [isson-netty-1-6] o.redisson.connection.ConnectionsHolder  : 24 connections initialized for 127.0.0.1/127.0.0.1:1112
2025-01-19T11:44:19.966-05:00  INFO 9959 --- [demo] [sson-netty-1-15] o.redisson.connection.ConnectionsHolder  : 1 connections initialized for 127.0.0.1/127.0.0.1:1111
2025-01-19T11:44:19.966-05:00  INFO 9959 --- [demo] [sson-netty-1-14] o.redisson.connection.ConnectionsHolder  : 1 connections initialized for 127.0.0.1/127.0.0.1:1110
2025-01-19T11:44:20.130-05:00  INFO 9959 --- [demo] [sson-netty-1-10] o.redisson.connection.ConnectionsHolder  : 24 connections initialized for 127.0.0.1/127.0.0.1:1111
2025-01-19T11:44:20.130-05:00  INFO 9959 --- [demo] [sson-netty-1-11] o.redisson.connection.ConnectionsHolder  : 24 connections initialized for 127.0.0.1/127.0.0.1:1110
Mapping: redis://valkey-node-0.valkey-headless.default.svc.cluster.local:6379
Re-mapped: redis://localhost:1110
Mapping: redis://valkey-node-2.valkey-headless.default.svc.cluster.local:6379
Re-mapped: redis://localhost:1112
Mapping: redis://valkey-node-1.valkey-headless.default.svc.cluster.local:6379
Re-mapped: redis://localhost:1111
Mapping: redis://valkey-node-2.valkey-headless.default.svc.cluster.local:6379
Re-mapped: redis://localhost:1112
Mapping: redis://valkey-node-1.valkey-headless.default.svc.cluster.local:26379
Re-mapped: redis://localhost:2111
Mapping: redis://valkey-node-0.valkey-headless.default.svc.cluster.local:26379
Re-mapped: redis://localhost:2110
2025-01-19T11:44:21.262-05:00 ERROR 9959 --- [demo] [sson-netty-1-13] org.redisson.connection.ServiceManager   : Unable to resolve redis://valkey-node-2.valkey-headless.default.svc.cluster.local:6379

io.netty.resolver.dns.DnsResolveContext$SearchDomainUnknownHostException: Failed to resolve 'valkey-node-2.valkey-headless.default.svc.cluster.local' [A(1)] and search domain query for configured domains failed as well: [home]
        at io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:1150) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsResolveContext.tryToFinishResolve(DnsResolveContext.java:1097) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsResolveContext.query(DnsResolveContext.java:455) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsResolveContext.onResponse(DnsResolveContext.java:686) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsResolveContext.access$500(DnsResolveContext.java:70) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsResolveContext$2.operationComplete(DnsResolveContext.java:513) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsQueryContext.trySuccess(DnsQueryContext.java:308) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsQueryContext.finishSuccess(DnsQueryContext.java:299) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.resolver.dns.DnsNameResolver$DnsResponseHandler.channelRead(DnsNameResolver.java:1491) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:107) ~[netty-codec-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:97) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.116.Final.jar:4.1.116.Final]
        at java.base/java.lang.Thread.run(Thread.java:840) ~[na:na]
Caused by: io.netty.resolver.dns.DnsErrorCauseException: Query failed with NXDOMAIN
        at io.netty.resolver.dns.DnsResolveContext.onResponse(..)(Unknown Source) ~[netty-resolver-dns-4.1.116.Final.jar:4.1.116.Final]

^Corg.redisson.client.RedisException: java.lang.InterruptedException
        at org.redisson.command.CommandAsyncService.get(CommandAsyncService.java:180)
        at org.redisson.RedissonObject.get(RedissonObject.java:98)
        at org.redisson.RedissonBucket.set(RedissonBucket.java:173)
        at com.example.demo.SentinelDemo.sayHello(SentinelDemo.java:50)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMethod.invoke(InitDestroyAnnotationBeanPostProcessor.java:457)
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:401)
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:219)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:423)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:336)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:289)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:334)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1122)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1093)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1030)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
        at com.example.demo.DemoApplication.main(DemoApplication.java:10)
Caused by: java.lang.InterruptedException
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:386)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
        at org.redisson.command.CommandAsyncService.get(CommandAsyncService.java:176)
        ... 29 more
```

### Additional information

_No response_

--- a/redisson/src/main/java/org/redisson/connection/SentinelConnectionManager.java
+++ b/redisson/src/main/java/org/redisson/connection/SentinelConnectionManager.java
@@ -508,7 +508,8 @@ public class SentinelConnectionManager extends MasterSlaveConnectionManager {
         RFuture<RedisURI> masterFuture = connection.async(1, cfg.getRetryInterval(), cfg.getTimeout(),
                                                             StringCodec.INSTANCE, masterHostCommand, cfg.getMasterName());
         return masterFuture
-                .thenCompose(u -> serviceManager.resolveIP(scheme, u))
+                .thenCompose(u -> resolveIP(u.getHost(), "" + u.getPort()))
+                .thenApply(this::toURI)
                 .thenCompose(newMaster -> {
                     RedisURI current = currentMaster.get();
                     if (!newMaster.equals(current)

